src/main/python/pre_fx_data.py

```diff
@@ -0,0 +1,269 @@
+        """
+        super(PreFxData, self).__init__()
+
+        self._stimulus_ontology: Optional[StimulusOntology] = None
+        self._qc_criteria: Optional[Dict] = None
+        self.data_set: Optional[EphysDataSet] = None
+        self.nwb_path: Optional[str] = None
+
+    def _notifying_setter(
+        self, 
+        attr_name: str, 
+        value: Any, 
+        on_set: pyqtSignal, 
+        on_unset: pyqtSignal,
+        send_value: bool = False
+    ):
+        """ Utility for a setter that emits Qt signals when the attribute in 
+        question changes state.
+
+        Parameters
+        ----------
+        attr_name :
+            identifies attribute to be set
+        value : 
+            set attribute to this value
+        on_set : 
+            emitted when the new value is not None
+        on_unset :
+            emitted when the new value is None
+        send_value : 
+            if True, the new value will be included in the emitted signal
+
+        """
+        setattr(self, attr_name, value)
+
+        if value is None:
+            on_unset.emit()
+        else:
+            if send_value:
+                on_set.emit(value)
+            else:
+                on_set.emit()
+
+    @property
+    def stimulus_ontology(self) -> Optional[StimulusOntology]:
+        return self._stimulus_ontology
+
+    @stimulus_ontology.setter
+    def stimulus_ontology(self, value: Optional[StimulusOntology]):
+        self._notifying_setter(
+            "_stimulus_ontology", 
+            value,
+            self.stimulus_ontology_set, 
+            self.stimulus_ontology_unset,
+            send_value=True
+        )
+
+    @property
+    def qc_criteria(self) -> Optional[Dict]:
+        return self._qc_criteria
+
+    @qc_criteria.setter
+    def qc_criteria(self, value: Optional[Dict]):
+        self._notifying_setter(
+            "_qc_criteria", 
+            value,
+            self.qc_criteria_set, 
+            self.qc_criteria_unset,
+            send_value=True
+        )
+
+    def set_default_stimulus_ontology(self):
+        self.load_stimulus_ontology_from_json(
+            StimulusOntology.DEFAULT_STIMULUS_ONTOLOGY_FILE
+        )
+
+    def set_default_qc_criteria(self):
+        self.load_qc_criteria_from_json(DEFAULT_QC_CRITERIA_FILE)
+
+    def load_stimulus_ontology_from_json(self, path: str):
+        """ Attempts to read a stimulus ontology file from a JSON. If 
+        successful (and other required data are already set), attempts to 
+        run the pre-fx pipeline
+
+        Parameters
+        ----------
+        path : 
+            load ontology from here
+
+        """
+
+        try:
+            with open(path, "r") as ontology_file:
+                ontology_data = json.load(ontology_file)
+            ontology = StimulusOntology(ontology_data)
+            
+            if self.nwb_path is not None and self.qc_criteria is not None:
+                self.run_pipeline(
```

Comment:
Would it be better if loading ontology would send a signal to the controller indicating that PreFx data has changed, which would trigger the ipfx run?

Reply:
I don't think that the controller should be involved in that logic. Because the ipfx run is so fast and because it can fail, it makes sense to me to treat it as part of the (re)loading process

Reply:
from in-person: leaving as is because we want the pipeline run to be atomic with the load.